IOPool/Streamer/src/FRDEventMessage.cc

```diff
@@ -27,22 +29,25 @@ FRDEventMsgView::FRDEventMsgView(void* buf)
       adler32_(0),
       crc32c_(0) {
   uint32* bufPtr = static_cast<uint32*>(buf);
-  version_ = *bufPtr;
-  // if the version number is rather large, then we assume that the true
-  // version number is one.  (In version one of the format, there was
-  // no version number in the data, and the run number appeared first.)
-  if (version_ >= 32) {
-    version_ = 1;
-  }
 
-  size_ = 0;
+  // In version one of the format, there was no version number in the data,
+  // and the run number (32-bit) appeared first.
+  // This format is no longer supported
+  version_ = *bufPtr | 0xffff;
 
-  // version number
-  if (version_ >= 2) {
-    size_ += sizeof(uint32);
-    ++bufPtr;
+  if (version_ < 2) {
+    throw cms::Exception("FRDEventMsgView") << "FRD version " << version_ << " is no longer supported";
   }
```

Comment:
How practical is this exception? I suppose it would be thrown only when reading version 1 data where the lowest 16 bits of the word are `0` or `1`. (in which case, should `version_ > 6` be made to throw as well? or throw `if (version_ >= 32)` that was used above to deduce the version being 1?)

Reply:
We could improve it to throw when lower 16-bit value is not [2,6]. Forward compatibility for higher versions is not needed. In addition, for [2,5] we can require flag bits to all be set to 0.

Reply:
What I suggested has been committed. If there are no more comments on this, I will resolve this one.